plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -48,11 +46,16 @@ public class CatalogServiceImpl implements CatalogService {
 
   private Map<String, Catalog> catalogMap = new HashMap<>();
 
+  private final Map<ConnectorType, StorageEngineFactory> connectorTypeStorageEngineFactoryMap;
+
   public static CatalogServiceImpl getInstance() {
     return INSTANCE;
   }
 
   private CatalogServiceImpl() {
+    connectorTypeStorageEngineFactoryMap = new HashMap<>();
+    connectorTypeStorageEngineFactoryMap.put(ConnectorType.PROMETHEUS,
```

Comment:
get connector type from `PrometheusStorageFactory` rather than hardcoding the relation again?

Reply:
ok